src/test/java/apartment/data/ReservationFileRepositoryTest.java

```diff
@@ -0,0 +1,130 @@
+package apartment.data;
+
+
+import apartment.models.Host;
+import apartment.models.Reservation;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.time.LocalDate;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ReservationFileRepositoryTest {
+
+    static final String SEED_FILE_PATH = "./data/seed-2e72f86c-b8fe-4265-b4f1-304dea8762db.csv";
+    static final String TEST_FILE_PATH = "./data/reservations_test/2e72f86c-b8fe-4265-b4f1-304dea8762db.csv";
+    static final String TEST_DIR_PATH = "./data/reservations_test";
+
+    static final int NUM_OF_RESERVATIONS = 12;
+
+    ReservationFileRepository repository = new ReservationFileRepository(TEST_DIR_PATH);
+
+    @BeforeEach
+    void setup() throws IOException {
+        Path seedPath = Paths.get(SEED_FILE_PATH);
+        Path testPath = Paths.get(TEST_FILE_PATH);
+        Files.copy(seedPath, testPath, StandardCopyOption.REPLACE_EXISTING);
+    }
+
+    @Test
+    void findAllReservationsInFile() {
+        List<Reservation> reservations = repository.findByHostId("2e72f86c-b8fe-4265-b4f1-304dea8762db");
+        assertEquals(NUM_OF_RESERVATIONS, reservations.size());
+    }
+
+    @Test
+    void findingAHostIdThatDoesNotExist() {
+        List<Reservation> reservations = repository.findByHostId("id-doesn'-exist-9fafk-jfklaj");
+        assertEquals(0, reservations.size());
+    }
+
+    @Test
+    void addReservation() throws DataException {
+        Reservation reservation = new Reservation();
+        Host host = new Host();
+        host.setId("2e72f86c-b8fe-4265-b4f1-304dea8762db");
+
+        reservation.setHost(host);
+        reservation.setStartDate(LocalDate.of(1998, 12, 1));
+        reservation.setEndDate(LocalDate.of(1998, 12, 29));
+        reservation.setTotal(new BigDecimal("1000.20"));
+
+        reservation = repository.add(reservation);
+        assertEquals(13, reservation.getId());
+    }
```

Comment:
After performing a create, update, or delete operation, it'd be good to retrieve all of the records and confirm that you have the expected number of records and for add/update that you have the expected values.